### PR TITLE
adds gql linter

### PR DESCRIPTION
**Pull request by redreceipt**
## ✏️ Description

Adds ESLint plugin to check for GQL syntax errors

## 🔬 To Test
1. Introduce a GQL error, watch it flag

## 📸 Screenshots

Screenshot 2023-02-01 at 10 16 42 AM

## 📝 Checklist

- [x] I have performed a self-review of my code
- [x] I have tested my code works in the project (nothing breaks)
- [x] I have commented my code in hard-to-understand areas
- [x] New and existing tests and linting pass locally with my changes
- [x] I have made corresponding changes to the documentation (new envs?)
- [x] For PR's with design updates/changes, I have consulted with the product designer and made all necessary adjustments in Figma

### CODE REVIEW

**Comment by vinnyjth** [comment]
🙌

**Review by vinnyjth — APPROVED** [review]

File: .eslintrc.js [hunk 1]
```diff
@@ -4,4 +4,18 @@ module.exports = {
   env: {
     'jest/globals': true,
   },
+  overrides: [
+    {
+      files: ['*.js'],
```
Comment (dsilva609) [inline .eslintrc.js:9]: Not sure if we're there yet but should we worry about Typescript files as well?

Comment (redreceipt) [reply .eslintrc.js:9]: Good call

Comment (redreceipt) [inline .eslintrc.js:9]:
```suggestion
      files: ['*.js', '*.ts', '*.tsx'],
```

File: templates/mobile/.eslintrc.js [hunk 2]
```diff
@@ -9,4 +9,18 @@ module.exports = {
   rules: {
     '@typescript-eslint/no-var-requires': 1,
   },
+  overrides: [
+    {
+      files: ['*.js'],
```
Comment (redreceipt) [inline templates/mobile/.eslintrc.js:14]:
```suggestion
      files: ['*.js', '*.ts', '*.tsx'],
```

File: templates/tv/.eslintrc.js [hunk 3]
```diff
@@ -4,4 +4,18 @@ module.exports = {
   env: {
     'jest/globals': true,
   },
+  overrides: [
+    {
+      files: ['*.js'],
```
Comment (redreceipt) [inline templates/tv/.eslintrc.js:9]:
```suggestion
      files: ['*.js', '*.ts', '*.tsx'],
```

**Review by dsilva609 — APPROVED** [review]

**Pull request merged**